rtos/TARGET_CORTEX/rtx5/RTX/Source/rtx_core_ca.h

```diff
@@ -112,6 +112,18 @@ static __asm    uint32_t __get_PSP (void) {
   sub   r0, r0, #32
   bx    lr
 }
+#elif (defined (__ARMCC_VERSION) && (__ARMCC_VERSION >= 6010050)) || defined ( __GNUC__ )
+__STATIC_INLINE __attribute__((naked)) uint32_t __get_PSP (void) {
+  __ASM volatile (
+    ".syntax unified\n\t"
+    ".arm\n\t"
+    "sub  sp,sp,#4\n\t"
+    "stm  sp,{sp}^\n\t"
+    "pop  {r0}\n\t"
+    "sub  r0,r0,#32\n\t"
+    "bx   lr\n\t"
+  );
+}
```

Comment:
This doesn't seem right for a few reasons.

* It's a change to upstream CMSIS-RTOS code
* It doesn't put the assembler state back to Thumb afterwards, if it was
* It doesn't mark the function itself as being ARM, rather than Thumb, so you'll be entered in Thumb state, if `-mthumb` is enabled. The assembler is being switched, not the CPU.

Now, looking at the original CMSIS-RTOS code, I think this header file doesn't actually cope with the compiler being set to Thumb at all for GCC or ARMC5, and that's a pre-existing issue - it's assuming the compiler is set to compile ARM.

I think we should probably side-step the issue for now by not attempting to use Thumb mode when compiling for A-profile. ARMC6 is out of line here. GCC and ARMC5 are both set to use ARM, and that's why this is failing for ARMC6.

I think the `-mthumb` should be removed from the ARMC6 profiles. I don't think you need to specify at all to match the other toolchains - the default is `-marm` if specifying a CPU that supports it, else `-mthumb`. If it did need specification, then it should be in `arm.py` like it is in `gcc.py`, where it's set based on CPU.

Reply:
Raising a separate PR to change the tool profiles.

Reply:
Thank you for review. I reverted it.

Reply:
It would be useful if you could confirm that each of https://github.com/ARMmbed/mbed-os/pull/9612 and https://github.com/ARM-software/CMSIS_5/pull/527 solve the issue on their own.